src/components/SearchForm.js

```diff
@@ -0,0 +1,74 @@
+import { useContext, useState } from 'react';
+import { GoogleMapsContext } from '../context/GoogleMapsContext';
+import { MapCenterContext } from '../context/MapCenterContext';
+import planetSearch from '../assets/search-location-solid.svg';
+
+export const SearchForm = () => {
+  const [query, setQuery] = useState('');
+  const valueGoogleMapsAPI = useContext(GoogleMapsContext);
+  const valueMapCenter = useContext(MapCenterContext);
+  const { map, maps } = valueGoogleMapsAPI;
+  const { setNewCenterMap } = valueMapCenter;
+
+  const handleQuery = (e) => setQuery(e.target.value);
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+
+    const request = {
+      query,
+      fields: ['name', 'geometry'],
+    };
+
+    const service = new maps.places.PlacesService(map);
+
+    service.findPlaceFromQuery(request, function (results, status) {
+      if (status === maps.places.PlacesServiceStatus.REQUEST_DENIED) {
+        alert(
+          'We are so sorry! Places API is not enabled. Please try again later.',
+        );
+      }
+      if (status === maps.places.PlacesServiceStatus.ZERO_RESULTS) {
```

Comment:
I wonder about using `else if` here or maybe even a `switch` statement. The way this is coded, you will still execute the other conditional checks even if the first one resolves to `true`

Reply:
Those small details are incredible! Thanks a million 🥳 😄 👍 😮 !